src/Services/QueuedJob.php

```diff
@@ -84,6 +84,23 @@ public function prepareForRestart();
      */
     public function getJobType();
 
+    /**
+     * Specifies what user ID should be when running the job
+     * valid values:
+     * null - (default) - run the job as current user
+     * 0 - run the job without a user
+     * greater than zero - run the job as a specific user
+     *
+     * This is useful in situations like:
+     * - a job needs to always run without a user (like a static cache job)
+     * - a job needs to run as a specific user (for example data migration job)
+     *
+     * Note that this value can be overriden in the @see QueuedJobService::queueJob()
+     *
+     * @return int|null
+     */
+    public function getRunAsMemberID();
```

Comment:
Sorry, I didn't pick this up as a breaking change. While AbstractQueuedJob implements this interface, not all jobs do (including those in this module). We'll need to remove this from the interface and adjust the call logic to do a `method_exists()` check

Reply:
@robbieaverill I'm not sure I follow. As far as I can see, all jobs in this module extend the `AbstractQueuedJob` which has the function defined and uses the interface. COuld you please clarify what you mean by breaking change exactly?

Reply:
Yeah sorry, it's not in this module but in fulltextsearch - see `SearchUpdateCommitJobProcessor` as an example that implements the interface but doesn't extend the abstract base class

Reply:
Unfortunately @mfendeksilverstripe adding methods to interfaces is a breaking change :(.

Reply:
You may introduce new interfaces though

Reply:
I see I haven't checked other module. I like the idea of a new Interface. Should I create a PR for that, or are you going to hotfix it?

Reply:
We won't release until it's resolved. You're welcome to create a PR for it otherwise I'll try to get around to it at some point today.

Yeah we can't even introduce a method to an interface even if we go and update all the open source code to implement the new method. We can't see all the closed source code that might implement this interface too 😉

Reply:
I'll have a look at it today. It shouldn't be too much work.

Reply:
Open PR with the fix: https://github.com/symbiote/silverstripe-queuedjobs/pull/262